File: Java/BOJ/17425 약수의 합.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int num = 1000001;
        long[] fn = new long[num];
        long[] gn = new long[num];
        for (int i = 0; i < num; i++) {
        	fn[i] = 1;
        }
        for (int i = 2; i < num; i++) {
        	int j = 1;
        	while (i*j < num) {
        		fn[i*j] += i;
        		j++;
        	}
        }
        
        for (int i = 1; i < num; i++) {
        	gn[i] = gn[i-1] + fn[i];
        }
        int T = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();
        for (int i = 0 ; i < T; i++) {
        	int n = Integer.parseInt(br.readLine());
        	sb.append(gn[n]).append('\n');
        }
        System.out.println(sb);
    }
}

File: Python/BOJ/17425 약수의 합.py
import sys
num = 1000001
fn = [1] * (num)
gn= [0]*(num)

for i in range(2, num):
    j = 1
    while i*j < num:
        fn[i*j] += i
        j += 1

for i in range(1, num):
    gn[i] = gn[i-1] + fn[i]

T = int(sys.stdin.readline())
for _ in range(T):
    n = int(sys.stdin.readline())
    print(gn[n])